frontend/src/App.js

```diff
@@ -44,23 +50,32 @@ function App() {
 
   return (
     <UserContext.Provider value={user}>
-      {
-        user ?
-          <>
-            <Nav onLogOut={handleLogOut}/>
-            <Routes>
-              <Route path="/dashboard/" element={<Dashboard/>}/>
-              <Route path="/markets" element={<Market/>}/>
-              <Route path="*" element={<Navigate to="/dashboard"/>}/>
-            </Routes>
-          </>
-          :
+      {user ? (
+        <>
```

Comment:
```
{(
    <>
      <Nav onLogOut={handleLogOut} />
      <Routes>
       {user ? 
        (
        <>
          <Route path="/dashboard/" element={<Dashboard />} />
          <Route path="/markets" element={<Market />} />
          <Route path="*" element={<NotFound />} />
        </>
        ) : 
        (
        <>
         <Route
            path="/signin"
            element={
              <SignInUpForm isSignIn={true} onSuccess={onSignInSuccess} />
            }
          />
          <Route
            path="/signup"
            element={
              <SignInUpForm isSignIn={false} onSuccess={onSignUpSuccess} />
            }
          />
        </>
        )
      }
      </Routes>
  }
    </>
  )
}
```

This avoids repeating common routes for both branches, in this case, 404. Haven't used react-router for a while but I think this should work.

Reply:
Yes, that's the idea. Do verify it before merging, though. There might be caveats.